src/types/params/plain_secret.rs

```diff
@@ -389,3 +470,39 @@ fn rsa_secret_params(i: &[u8]) -> IResult<&[u8], PlainSecretParams> {
         }
     })(i)
 }
+
+/// Derive output keying material and associated data for the s2k usage method AEAD.
+///
+/// https://www.rfc-editor.org/rfc/rfc9580.html#name-secret-key-packet-formats
+pub(crate) fn s2k_usage_aead(
+    derived: &[u8],
+    secret_tag: Tag,
+    pub_key: &(impl PublicKeyTrait + Serialize),
+    sym_alg: SymmetricKeyAlgorithm,
+    aead_mode: AeadAlgorithm,
+) -> Result<([u8; 32], Vec<u8>)> {
+    // HKDF to derive output keying material
+    let hk = Hkdf::<Sha256>::new(None, derived);
+    let mut okm = [0u8; 32];
+
+    let type_id = u8::from(secret_tag) | 0xc0;
+
+    // HKDF info parameter
+    let info = &[
```

Comment:
this is a bit odd, I would skip the reference here, and only use it at call site